tests/cpp/tree/test_gpu_hist_sampling.cc

```diff
@@ -0,0 +1,128 @@
+/*!
+ * Copyright 2020 XGBoost contributors
+ */
+#include <dmlc/filesystem.h>
+#include <fstream>
+#include <xgboost/base.h>
+#include <xgboost/data.h>
+#include <xgboost/learner.h>
+
+#include "gtest/gtest.h"
+
+namespace xgboost {
+namespace tree {
+
+class GpuHistSamplingTest : public ::testing::Test {
+ protected:
+  static void SetUpTestCase() {
+    constexpr size_t kRows = 1000;
+    constexpr size_t kCols = 1;
+    constexpr size_t kPageSize = 1024;
+
+    temp_dir = new dmlc::TemporaryDirectory();
+    const std::string tmp_file = temp_dir->path + "/random.libsvm";
+    {
+      std::ofstream fo(tmp_file.c_str());
+
+      std::mt19937 gen{2020}; // NOLINT
+      std::normal_distribution<> rnd{0, 1};
+
+      for (size_t i = 0; i < kRows; i++) {
+        std::stringstream row;
+        row << rnd(gen);
+
+        for (size_t j = 0; j < kCols; j++) {
+          row << " " << j << ":" << rnd(gen);
+        }
+        fo << row.str() << "\n";
+      }
+    }
+
+    dmat = std::shared_ptr<DMatrix>(DMatrix::Load(tmp_file, true, false));
+    const std::string ext_mem_file = tmp_file + "#" + tmp_file + ".cache";
+    dmat_ext = std::shared_ptr<DMatrix>(
+        DMatrix::Load(ext_mem_file, true, false, "auto", kPageSize));
+  }
+
+  static void TearDownTestCase() {
+    dmat.reset();
+    dmat_ext.reset();
+    delete temp_dir;
+  }
+
+  static void VerifyPredictionMean(const std::shared_ptr<DMatrix>& dtrain,
```

Comment:
@RAMitchell is this close to what you had in mind to verify samples provide unbiased estimate?